### PR TITLE
Add JsonSerializerSettings to NewtonsoftJsonPayloadSerializer for the hub device

### PR DESCRIPTION
**Pull request by brycewang-microsoft**
We have the customizable Newtonsoft.Json serializer for the hub device in v2. This is to fix the similar date-time string problem like this [one](https://github.com/Azure/azure-iot-sdk-csharp/pull/3097).

### CODE REVIEW

File: iothub/service/src/JsonSerializerSettingsInitializer.cs [hunk 1]
```diff
@@ -9,7 +9,7 @@ namespace Microsoft.Azure.Devices
     /// <summary>
     /// A class to initialize JsonSerializerSettings which can be applied to the project.
     /// </summary>
-    public static class JsonSerializerSettingsInitializer
+    internal static class JsonSerializerSettingsInitializer
```
Comment (brycewang-microsoft) [inline iothub/service/src/JsonSerializerSettingsInitializer.cs:12]: As per our discussion, change the access to internal

Comment (abhipsaMisra) [reply iothub/service/src/JsonSerializerSettingsInitializer.cs:12]: The access modifiers for the methods down below can be made internal as well.

**Review by abhipsaMisra — APPROVED** [review]

File: iothub/device/src/Serialization/NewtonsoftJsonPayloadSerializer.cs [hunk 2]
```diff
@@ -18,14 +19,27 @@ public class NewtonsoftJsonPayloadSerializer : PayloadSerializer
...
 
+        /// <summary>
+        /// A static instance of JsonSerializerSettings which sets DateParseHandling to None.
+        /// </summary>
+        /// <remarks>
+        /// By default, serializing/deserializing with Newtonsoft.Json will try to parse date-formatted
+        /// strings to a date type, which drops trailing zeros in the microseconds date portion. By
+        /// specifying DateParseHandling with None, the original string will be read as-is.
+        /// </remarks>
+        public static readonly JsonSerializerSettings Settings = new()
```
Comment (drwill-ms) [inline iothub/device/src/Serialization/NewtonsoftJsonPayloadSerializer.cs:38]: Why public?

File: iothub/device/src/Serialization/NewtonsoftJsonPayloadSerializer.cs [hunk 3]
```diff
@@ -88,5 +102,13 @@ internal static bool TryGetNestedJsonObjectValue<T>(JObject nestedJsonObject, st
 
             return false;
         }
+
+        /// <summary>
+        /// Returns JsonSerializerSettings Func delegate
+        /// </summary>
+        internal static Func<JsonSerializerSettings> GetJsonSerializerSettingsDelegate()
```
Comment (drwill-ms) [inline iothub/device/src/Serialization/NewtonsoftJsonPayloadSerializer.cs:109]: Can this be private?

File: iothub/device/tests/NewtonsoftJsonPayloadSerializerTests.cs [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+{
+    [TestClass]
+    [TestCategory("UnitTest")]
+    public class NewtonsoftJsonPayloadSerializerTests
+    {
+        private static readonly string s_dateTimeString = "2023-01-31T10:37:08.4599400";
+        private static readonly string s_serializedPayloadString = "{\"time\":\"2023-01-31T10:37:08.4599400\"}";
+
+        [TestMethod]
+        public void NewtonsoftJsonPayloadSerializer_SerializeToString()
```
Comment (drwill-ms) [inline iothub/device/tests/NewtonsoftJsonPayloadSerializerTests.cs:19]:
```suggestion
        public void NewtonsoftJsonPayloadSerializer_DateTime_RoundtripSerialization()
```

File: iothub/device/tests/NewtonsoftJsonPayloadSerializerTests.cs [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.Devices.Client.Tests
+{
+    [TestClass]
+    [TestCategory("UnitTest")]
+    public class NewtonsoftJsonPayloadSerializerTests
```
Comment (drwill-ms) [inline iothub/device/tests/NewtonsoftJsonPayloadSerializerTests.cs:13]:
```suggestion
    /// <summary>
    /// Ensures DateTime de/serializes properly using Newtonsoft.Json, avoiding a known bug
    /// </summary>
    public class NewtonsoftJsonPayloadSerializerTests
```

Comment (drwill-ms) [reply iothub/device/tests/NewtonsoftJsonPayloadSerializerTests.cs:13]: Can you link the known bug?

File: iothub/device/tests/NewtonsoftJsonPayloadSerializerTests.cs [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+            // act
+            var result = NewtonsoftJsonPayloadSerializer.Instance.SerializeToString(testDateTime);
+
+            // assert
+            result.Should().Be(s_serializedPayloadString);
+        }
+
+        [TestMethod]
+        public void NewtonsoftJsonPayloadSerializer_DeserializeToType()
```
Comment (drwill-ms) [inline iothub/device/tests/NewtonsoftJsonPayloadSerializerTests.cs:32]:
```suggestion
        public void NewtonsoftJsonPayloadSerializer_DateTime_DeserializesProperly()
```

File: iothub/device/src/Serialization/NewtonsoftJsonPayloadSerializer.cs (hunk 2 above)
Comment (brycewang-microsoft) [reply iothub/device/src/Serialization/NewtonsoftJsonPayloadSerializer.cs:38]: will change it and below to internal

**Review by drwill-ms — APPROVED** [review]

**Pull request merged**